kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -1521,39 +1518,41 @@ impl<'tcx> GotocCtx<'tcx> {
         &mut self,
         f: F,
         mut fargs: Vec<Expr>,
-        p: &Place<'tcx>,
-        span: Option<Span>,
-        rust_arg_types: &[Ty<'tcx>],
-        rust_ret_type: Ty<'tcx>,
+        p: &Place,
+        span: Span,
+        rust_arg_types: &[Ty],
+        rust_ret_type: Ty,
     ) -> Stmt {
         let arg1 = fargs.remove(0);
         let arg2 = fargs.remove(0);
-        let ret_typ = self.codegen_ty(rust_ret_type);
+        let ret_typ = self.codegen_ty_stable(rust_ret_type);
 
         if arg1.typ().len().unwrap() != ret_typ.len().unwrap() {
             let err_msg = format!(
                 "expected return type with length {} (same as input type `{}`), \
                 found `{}` with length {}",
                 arg1.typ().len().unwrap(),
-                rust_arg_types[0],
-                rust_ret_type,
+                pretty_ty(rust_arg_types[0]),
+                pretty_ty(rust_ret_type),
                 ret_typ.len().unwrap()
             );
-            self.tcx.sess.span_err(span.unwrap(), err_msg);
+            span_err(self.tcx, span, err_msg);
         }
 
         if !ret_typ.base_type().unwrap().is_integer() {
-            let (_, rust_base_type) = rust_ret_type.simd_size_and_type(self.tcx);
+            let (_, rust_base_type) = self.simd_size_and_type(rust_ret_type);
             let err_msg = format!(
-                "expected return type with integer elements, found `{rust_ret_type}` with non-integer `{rust_base_type}`",
+                "expected return type with integer elements, found `{}` with non-integer `{}`",
+                pretty_ty(rust_ret_type),
+                pretty_ty(rust_base_type),
```

Comment:
Do we have plans to make `pretty_ty` the implementation for `Display`?

Reply:
We haven't discussed the implementation of Display yet. I think it makes sense to have Display being the user friendly version of things though. I created https://github.com/rust-lang/project-stable-mir/issues/57 to capture that request.